src/sources/soundsourceproxy.cpp

```diff
@@ -155,6 +166,19 @@ bool SoundSourceProxy::isFileExtensionSupported(const QString& fileExtension) {
     return !s_soundSourceProviders.getRegistrationsForFileExtension(fileExtension).isEmpty();
 }
 
+//static
+mixxx::SoundSourceProviderPointer SoundSourceProxy::getPrimaryProviderForFileExtension(
+        const QString& fileExtension) {
+    const auto optProviderRegistration =
+            s_soundSourceProviders.getPrimaryRegistrationForFileExtension(
+                    fileExtension);
+    if (optProviderRegistration) {
+        return optProviderRegistration->getProvider();
+    } else {
+        return {};
```

Comment:
```suggestion
        return nullptr;
```

?

No sure, but looks more explicit to me.